flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/TemporalRowtimeJoin.scala

```diff
@@ -237,15 +238,17 @@ class TemporalRowtimeJoin(
     * be joined with it later
     */
   private def cleanUpState(timerTimestamp: Long, rightRowsSorted: util.List[Row]) = {
-    for (i <- 0 until firstIndexToKeep(timerTimestamp, rightRowsSorted)) {
+    var i = 0
+    while (i < firstIndexToKeep(timerTimestamp, rightRowsSorted)) {
```

Comment:
Call `firstIndexToKeep()` only once.

Reply:
Ok changed, however, it shouldn't matter here.